docs/updating/migration-to-cdn/vuejs-v3.md

```diff
@@ -0,0 +1,189 @@
+	toolbar: [ 'heading', '|', 'bold', 'italic' ]
+} );
+</script>
+```
+
+**After:**
+
+```html
+<template>
+	<h2>Using CKEditor 5 from NPM in Vue 3</h2>
+  	<ckeditor
+		v-if="TestEditor"
+		v-model="data"
+		tag-name="textarea"
+		:editor="TestEditor"
+		:config="config"
+	/>
+</template>
+
+<script setup lang="ts">
+import { ref, reactive } from 'vue';
+import { useCKEditorCloud } from '@ckeditor/ckeditor5-vue';
+
+const cloud = useCKEditorCloud( {
+	version: '{@var ckeditor5-version}'
+} );
+
+const TestEditor = computed( () => {
+	if ( !cloud.data.value ) {
+		return null;
+	}
+
+	const {
+		ClassicEditor: BaseEditor, Paragraph,
+		Essentials, Heading, Bold, Italic
+	} = cloud.data.value.CKEditor;
+
+	return class TestEditor extends BaseEditor {
+		static builtinPlugins = [
+			Essentials,
+			Paragraph,
+			Heading,
+			Bold,
+			Italic
+		];
+	};
+} );
+
+const data = ref( '<p>Hello world!</p>' );
+const config = reactive( {
+	toolbar: [ 'heading', '|', 'bold', 'italic' ]
+} );
+</script>
+```
+
+### Step 3 (Optional): Migrate the CKEditor&nbsp;5 Vue 3+ integration testing suite
+
+If you have any tests that use CKEditor&nbsp;5 objects, you need to update them to use the  `loadCKEditorCloud` function. Here is an example of migrating a test that uses the `ClassicEditor` object:
+
+**Before:**
+
+```javascript
+import { ClassicEditor, ... } from 'ckeditor5';
+
+it( 'ClassicEditor test', () => {
+	// Your test that uses the CKEditor 5 object.
+} );
+```
+
+**After:**
+
+```javascript
+// It may be counterintuitive that in tests you need to use `loadCKEditorCloud` instead of `useCKEditorCloud`.
+// The reason for this is that `useCKEditorCloud` is a composable and can only be used in Vue components,
+// while tests are typically written as functions in testing suites. Therefore, in tests, you should use
+// the `loadCKEditorCloud` function to load CKEditor 5 from the CKEditor Cloud and obtain the necessary
+// CKEditor 5 objects. This allows you to properly test your CKEditor 5 integration without any issues.
+
+import { loadCKEditorCloud } from '@ckeditor/ckeditor5-vue';
+
+let cloud;
+
+beforeEach( async () => {
+	cloud = await loadCKEditorCloud( {
+		version: '{@var ckeditor5-version}',
+	} );
+} );
+
+it( 'ClassicEditor test', () => {
+	const { ClassicEditor, ... } = cloud.CKEditor;
+
+	// Your test that uses the CKEditor 5 object.
+} );
+```
+
+#### Step 4 (Optional): Clean up the document head entries before each test
+
+The `useCKEditorCloud` composable under the hood injects the CKEditor&nbsp;5 scripts and styles into your document head. If you are using a testing suite that does not Clean up the document head entries before each test, you may need to do it manually. This is important because the `useCKEditorCloud` composable might reuse the same head entries for each test, which can lead to skipping `loading` state and directly going to the `success` state. It may cause some tests that rely on the `loading` state to fail.
```

Comment:
```suggestion
The `useCKEditorCloud` composable under the hood injects the CKEditor&nbsp;5 scripts and styles into your document head. If you use a testing suite that does not Clean up the document head entries before each test, you may need to do it manually. This is important because the `useCKEditorCloud` composable might reuse the same head entries for each test, which can lead to skipping the `loading` state and directly going to the `success` state. It may cause some tests that rely on the `loading` state to fail.
```